go/vt/topo/etcd2topo/server.go

```diff
@@ -71,12 +82,42 @@ func (s *Server) Close() {
 	s.cli = nil
 }
 
-// NewServer returns a new etcdtopo.Server.
-func NewServer(serverAddr, root string) (*Server, error) {
-	cli, err := clientv3.New(clientv3.Config{
-		Endpoints:   strings.Split(serverAddr, ","),
+// TODO: Rename this to NewServer and change NewServer to a name that signifies it's global.
+func NewServerWithOpts(serverAddr, root string, certPath *string, keyPath *string, caPath *string) (*Server, error){
+	config := clientv3.Config{
+		Endpoints:  strings.Split(serverAddr, ","),
 		DialTimeout: 5 * time.Second,
-	})
+	}
+
+	// If TLS is enabled, attach TLS config info.
+	if *certPath != "" && *keyPath != "" {
+		// Verify we can load provided cert and key.
+		_, err := tls.LoadX509KeyPair(*certPath, *keyPath)
+		if err != nil {
+			log.Fatalf("Unable to load cert %v and key %v, err %v", *certPath, *keyPath, err)
```

Comment:
Just return an error instead of crashing.

Reply:
Actually if we're not using the result of these load/read attempts, let's just let the client lib return an error if it can't read these files. We don't need to double-check.

Reply:
Alright, noted.

Reply:
Done https://github.com/vitessio/vitess/pull/5496/commits/805108d2a52fbbe3c4145e3225f3fe93e81d0e21.